internal/controller/snapshot/snapshot_adapter.go

```diff
@@ -853,6 +858,23 @@ func (a *Adapter) updateComponentSource(ctx context.Context, c client.Client, co
 	return nil
 }
 
+func (a *Adapter) updateComponentLastPromotedImage(ctx context.Context, c client.Client, component *applicationapiv1alpha1.Component, snapshotComponent *applicationapiv1alpha1.SnapshotComponent) error {
+	if component.Status.LastPromotedImage != snapshotComponent.ContainerImage {
```

Comment:
would it be optimal if we check first in case of same image, then early exit, wdyt?
```
  if component.Status.LastPromotedImage == snapshotComponent.ContainerImage {
    return nil
  }
```